homeassistant/components/notify/__init__.py

```diff
@@ -63,3 +76,60 @@ async def persistent_notification(service: ServiceCall) -> None:
     )
 
     return True
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up a config entry."""
+    service_integration: ServiceIntegration = hass.data[DOMAIN]
+    return await service_integration.async_setup_entry(entry)
+
+
+async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Unload a config entry."""
+    service_integration: ServiceIntegration = hass.data[DOMAIN]
+    return await service_integration.async_unload_entry(entry)
+
+
+class NotifyService(PlatformService):
+    """Represent a notify platform service."""
+
+    @property
+    def service_name(self) -> str:
+        """Return the name of the service, not including domain."""
+        return f"{self.platform.platform_name}_{SERVICE_NOTIFY}"
+
+    @property
+    def service_description(self) -> ServiceDescription:
```

Comment:
I guess one thing I realized now, and I am not sure if it’s a problem, but this limits each integration to providing a single service.

Reply:
I don't think so. Each PlatformService class represents one service kind. Notify could define more than one PlatformService classes if it wants different services.

If an implementing platform wants to register different versions of the same PlatformService class it can overwrite the service description and return different descriptions for different instances.